typing/typecore.mli

```diff
@@ -142,6 +142,7 @@ type error =
   | Virtual_class of Longident.t
   | Private_type of type_expr
   | Private_label of Longident.t * type_expr
+  | Private_constructor_extension of constructor_description * type_expr
```

Comment:
May I suggest  `Private_extension_constructor`  rather than `Private_constructor_extension` which sounds a bit like an extension of a private constructor?

Reply:
We could also be future proof and name it `Private_constructor`.

Reply:
Renamed to `Private_extension_constructor`

Reply:
I agree with @trefis on that one.

Reply:
Exception renamed to `Private_constructor`